tests/test_masterclasses.py

```diff
@@ -97,6 +97,24 @@ def test_logging_in(test_client, db, test_user):
     assert response.location == f'http://localhost{url_for("main_bp.index")}'
 
 
+@pytest.mark.parametrize(
```

Comment:
I love this! It's going to be moderately annoying to scale though...I wonder if there's a way of automatically pulling all the protected/unprotected routes?

Not a question you need an immediate answer to and certainly not one I know the answer to!

Reply:
A very good point! I wonder if a for loop for each login required route would work. I will think about this more in the my next PRs as the number of routes grows.